rust/src/tantivy_util/util.rs

```diff
@@ -7,33 +7,50 @@ use tantivy::{Index, Term};
 
 pub const DOCUMENT_BUDGET_BYTES: usize = 50_000_000;
 
-pub fn extract_text_from_owned_value<'a>(value: &'a OwnedValue) -> Option<Cow<'a, str>> {
-    match value {
-        OwnedValue::Str(text) => Some(Cow::Borrowed(text)),
-        _ => None,
+pub fn extract_text_from_owned_value<'a>(
+    value: &'a OwnedValue,
+) -> Result<Cow<'a, str>, TantivyGoError> {
+    if let OwnedValue::Str(text) = value {
+        Ok(Cow::Borrowed(text))
+    } else {
+        Err(TantivyGoError("Only OwnedValue::Str is supported".to_string()))
     }
 }
 
 pub fn extract_terms(
     index: &Index,
     field: Field,
     query: &str,
-) -> Result<Vec<(usize, Term)>, Box<dyn Error>> {
-    let mut tokenizer = index.tokenizer_for_field(field)?;
+) -> Result<Vec<(usize, Term)>, TantivyGoError> {
+    let mut tokenizer = match index.tokenizer_for_field(field) {
+        Ok(tokenizer) => tokenizer,
+        Err(err) => return Err(TantivyGoError::from_err("", &err.to_string())),
+    };
     let mut token_stream = tokenizer.token_stream(query);
     let mut terms = Vec::new();
-    token_stream
-        .process(&mut |token: &Token| terms.push((token.position, Term::from_field_text(field, &token.text))));
+    token_stream.process(&mut |token: &Token| {
+        terms.push((token.position, Term::from_field_text(field, &token.text)))
+    });
     if terms.len() > 0 {
         Ok(terms)
     } else {
-        Err("Zero terms were extracted".into())
+        Err(TantivyGoError("Zero terms were extracted".to_string()))
     }
 }
 
 #[derive(Debug)]
 pub struct TantivyGoError(pub String);
 
+impl TantivyGoError {
+    pub fn from_err(message: &str, err: &str) -> TantivyGoError {
```

Comment:
as I understood, `from_err` is mostly used to chain errors together. If that's the case, we can use https://docs.rs/error-chain/latest/error_chain/ in future for this purpose. 

Some other usefull error organizing crates:

- https://docs.rs/anyhow/latest/anyhow/ -- idiomatic error handing for rust, great for converting errors back and forth 
- https://docs.rs/thiserror/1.0.11/thiserror/ -- convenient tool to creating error enums